lib/validate/validate_property.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+var validate = require('./validate');
+var ValidationError = require('../error/validation_error');
+var getType = require('../util/get_type');
+
+module.exports = function validateProperty(options, propertyType) {
+    var key = options.key;
+    var style = options.style;
+    var styleSpec = options.styleSpec;
+    var value = options.value;
+    var propertyKey = options.objectKey;
+    var layerSpec = styleSpec[propertyType + '_' + options.layerType];
+
+    if (!layerSpec) return [];
+
+    var transitionMatch = propertyKey.match(/^(.*)-transition$/);
+    if (propertyType == 'paint' && transitionMatch && layerSpec[transitionMatch[1]] && layerSpec[transitionMatch[1]].transition) {
+        return validate({
+            key: key,
+            value: value,
+            valueSpec: styleSpec.transition,
+            style: style,
+            styleSpec: styleSpec
+        });
+    }
+
+    var valueSpec = options.valueSpec || layerSpec[propertyKey];
+    if (!valueSpec) {
+        return [new ValidationError(key, value, 'unknown property "%s"', propertyKey)];
+    }
+
+    var tokenMatch;
+    if (getType(value) === 'string' && valueSpec['property-function'] && !valueSpec.tokens && (tokenMatch = /^{([^}]+)}$/.exec(value))) {
+        return [new ValidationError(key, value, '"%s" does not support interpolation syntax\n' +
+            'Use an identity property function instead: `{ "type": "identity", "property": %s` }`.',
+            propertyKey, JSON.stringify(tokenMatch[1]))];
```

Comment:
The follow-on error (usually "expected ___, got string") doesn't add much additional value and could be confusing.